MicrosoftSentinel/microsoft_sentinel/connector_microsoft_sentinel.py

```diff
@@ -0,0 +1,190 @@
+import time
+import datetime as date_time
+from datetime import datetime, timedelta, timezone
+from dateutil.parser import isoparse
+
+import orjson
+import itertools
+from typing import Dict, Any, Callable, Optional
+from functools import cached_property
+
+from urllib3.exceptions import HTTPError as BaseHTTPError
+from requests.exceptions import HTTPError
+
+from azure.identity import ClientSecretCredential
+from azure.mgmt.securityinsight import SecurityInsights
+from azure.core.paging import ItemPaged
+from azure.mgmt.securityinsight.models import IncidentAdditionalData, IncidentOwnerInfo, Incident, IncidentLabel
+
+from sekoia_automation.connector import Connector
+from sekoia_automation.storage import PersistentJSON
+
+from microsoft_sentinel.utils import additional_data_to_dict, owner_data_to_dict, labels_data_to_dict
+from microsoft_sentinel.metrics import EVENTS_LAG, FORWARD_EVENTS_DURATION, INCOMING_EVENTS, OUTCOMING_EVENTS
+from microsoft_sentinel import (
+    MicrosoftSentinelModule,
+    MicrosoftSentinelResponseModel,
+    MicrosoftSentinelConnectorConfiguration,
+)
+
+
+class MicrosoftSentineldConnector(Connector):
+    module: MicrosoftSentinelModule
+    configuration: MicrosoftSentinelConnectorConfiguration
+
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs)
+        self.context = PersistentJSON("context.json", self._data_path)
+
+    @property
+    def checkpoint(self) -> str:
+        now = datetime.now(timezone.utc)
+
+        with self.context as cache:
+            cached_date = cache.get("cached_date")
+
+            if cached_date is None:
+                one_day_ago = now - timedelta(days=1)
+                return one_day_ago.strftime("%Y-%m-%dT%H:%M:%SZ")
+
+            cached_datetime: datetime = isoparse(cached_date)
+
+            one_month_ago: datetime = now - timedelta(days=30)
+            return max(cached_datetime, one_month_ago).strftime("%Y-%m-%dT%H:%M:%SZ")
+
+    @checkpoint.setter
+    def checkpoint(self, last_message_date: datetime) -> None:
+        with self.context as cache:
+            cache["cached_date"] = last_message_date
```

Comment:
Can you use [CheckpointDatetime](https://github.com/SEKOIA-IO/sekoia-automation-sdk/blob/main/sekoia_automation/checkpoint.py#L34) instead? (see [TrendMicro VisionOne](https://github.com/SEKOIA-IO/automation-library/blob/main/TrendMicro/trendmicro_modules/trigger_vision_one_oat.py) for inspiration)